Sources/StreamVideoSwiftUI/CallView/CallView.swift

```diff
@@ -7,96 +7,70 @@ import SwiftUI
 import StreamWebRTC
 
 public struct CallView<Factory: ViewFactory>: View {
-    
+
     @Injected(\.streamVideo) var streamVideo
     @Injected(\.images) var images
     @Injected(\.colors) var colors
-    
+
     private let padding: CGFloat = 16
-    
+
     var viewFactory: Factory
     @ObservedObject var viewModel: CallViewModel
-    
+
     public init(viewFactory: Factory, viewModel: CallViewModel) {
         self.viewFactory = viewFactory
         self.viewModel = viewModel
     }
-    
-    public var body: some View {
-        GeometryReader { reader in
-            ZStack {
-                VStack(spacing: 0) {
-                    GeometryReader { videoFeedProxy in
-                        if viewModel.localVideoPrimary {
-                            localVideoView
-                                .edgesIgnoringSafeArea(.top)
-                                .accessibility(identifier: "localVideoView")
-                        } else if let screenSharingSession = viewModel.call?.state.screenSharingSession,
-                                    viewModel.call?.state.isCurrentUserScreensharing == false {
-                            viewFactory.makeScreenSharingView(
-                                viewModel: viewModel,
-                                screensharingSession: screenSharingSession,
-                                availableFrame: videoFeedProxy.frame(in: .global)
-                            )
-                        } else {
-                            participantsView(bounds: videoFeedProxy.frame(in: .global))
-                        }
-                    }
 
-                    viewFactory.makeCallControlsView(viewModel: viewModel)
-                        .opacity(viewModel.hideUIElements ? 0 : 1)
-                }
+    public var body: some View {
+        VStack(spacing: 16) {
+            viewFactory.makeCallTopView(viewModel: viewModel)
 
-                VStack(alignment: .trailing, spacing: padding) {
```

Comment:
can you explain the changes in a nutshell here?

Reply:
They are mainly organisational in order to reduce the information from body and move it somewhere that the context can help disambiguate the usage.